File: backend/app/schemas/lms.py
from datetime import datetime
from decimal import Decimal
from typing import Optional
from pydantic import BaseModel


class CourseCreate(BaseModel):
    Title: str
    Slug: str
    ShortDescription: Optional[str] = None
    Description: Optional[str] = None
    Thumbnail: Optional[str] = None
    Price: Decimal = Decimal("0")
    Level: str = "Beginner"
    Status: str = "draft"
    InstructorID: Optional[int] = None
    IsPublished: bool = False


class CourseUpdate(BaseModel):
    Title: Optional[str] = None
    Slug: Optional[str] = None
    ShortDescription: Optional[str] = None
    Description: Optional[str] = None
    Thumbnail: Optional[str] = None
    Price: Optional[Decimal] = None
    Level: Optional[str] = None
    Status: Optional[str] = None
    InstructorID: Optional[int] = None
    IsPublished: Optional[bool] = None


class CourseResponse(CourseCreate):
    CourseID: int
    CreatedAt: datetime

    class Config:
        from_attributes = True


class LessonCreate(BaseModel):
    Title: str
    Content: Optional[str] = None
    VideoURL: Optional[str] = None
    ResourceURL: Optional[str] = None
    SortOrder: int = 0
    IsPreview: bool = False


class LessonResponse(LessonCreate):
    LessonID: int
    CourseID: int

    class Config:
        from_attributes = True


class EnrollmentCreate(BaseModel):
    StudentID: int
    CourseID: int


class AssignmentCreate(BaseModel):
    CourseID: int
    Title: str
    Instructions: Optional[str] = None
    DueAt: Optional[datetime] = None
    MaxMarks: Decimal = Decimal("100")


class SubmissionCreate(BaseModel):
    AnswerText: Optional[str] = None
    AttachmentURL: Optional[str] = None


class GradeSubmission(BaseModel):
    Marks: Decimal
    Feedback: Optional[str] = None

File: frontend/types/lms.ts
export type Course = {
  CourseID: number;
  Title: string;
  Slug: string;
  ShortDescription?: string | null;
  Description?: string | null;
  Thumbnail?: string | null;
  Price: number | string;
  Level: string;
  Status: string;
  InstructorID?: number | null;
  IsPublished: boolean;
  CreatedAt?: string;
};
export type Lesson = { LessonID:number; CourseID:number; Title:string; Content?:string|null; VideoURL?:string|null; ResourceURL?:string|null; SortOrder:number; IsPreview:boolean };
export type Enrollment = { EnrollmentID:number; StudentID:number; CourseID:number; Status:string; ProgressPercent:number; EnrolledAt?:string; student?:{UserID:number;FullName:string;Email:string}; course?:Course };
export type Assignment = { AssignmentID:number; CourseID:number; Title:string; Instructions?:string|null; DueAt?:string|null; MaxMarks:number|string; CreatedAt?:string; course?:Course };
export type Submission = { SubmissionID:number; AssignmentID:number; StudentID:number; AnswerText?:string|null; AttachmentURL?:string|null; Marks?:number|null; Feedback?:string|null; Status:string; SubmittedAt?:string; GradedAt?:string|null; student?:{UserID:number;FullName:string;Email:string}; assignment?:Assignment };
export type MyCourse = { enrollment_id:number; status:string; progress:number; course:Course };
